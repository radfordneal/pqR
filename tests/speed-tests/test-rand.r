source("time.r")
 
test.name <- "rand.unif"

for (gen in c("default","Wichmann-Hill"))
{ 
  set.seed(123,gen)

  f <- test.cmp(function() { for (i in 1:1000000) r <- runif(1); r })
  sys.time(a<-f())
  print(a)

  f <- test.cmp(function() { for (i in 1:1000000) r <- runif(10); r })
  sys.time(a<-f())
  print(a)

  f <- test.cmp(function() { for (i in 1:50000) r <- runif(1000); r })
  sys.time(a<-f())
  print(a[c(1,77,876,1000)])
}

set.seed(1,"default")

test.name <- "rand.norm"

for (ngen in c("default","Box-Muller"))
{ 
  set.seed(1234)
  RNGkind(normal.kind=ngen)

  f <- test.cmp(function() { for (i in 1:1000000) r <- rnorm(1); r })
  sys.time(a<-f())
  print(a)

  f <- test.cmp(function() { for (i in 1:1000000) r <- rnorm(10); r })
  sys.time(a<-f())
  print(a)

  f <- test.cmp(function() { for (i in 1:20000) r <- rnorm(1000); r })
  sys.time(a<-f())
  print(a[c(1,77,876,1000)])
}


# Correctness checks.

set.seed(2,"default")

print(runif(1))

a <- .Random.seed
print(sum(0+.Random.seed))

print(runif(1))

print(sum(0+a))
print(sum(0+.Random.seed))
