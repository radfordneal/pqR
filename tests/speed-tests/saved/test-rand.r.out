
---- rand.unif 

Time:
[1] 0.2297699
Time:
 [1] 0.1885836 0.7046904 0.4577854 0.5749424 0.9657214 0.5945703 0.1065796
 [8] 0.4610406 0.4808220 0.5922590
Time:
[1] 0.5281667 0.6734528 0.1851717 0.3281823
Time:
[1] 0.02805603
Time:
 [1] 0.31067052 0.06478731 0.96756429 0.09315947 0.24599851 0.22246476
 [7] 0.71294201 0.05110552 0.23349373 0.26200931
Time:
[1] 0.4076375 0.3045166 0.1221239 0.2359454

---- rand.norm 

Time:
[1] -0.04320838
Time:
 [1] -2.1447801 -0.2867886  2.7138008 -0.7872427 -0.8688311  0.3303366
 [7] -1.1634628 -0.4880991  1.8864387 -1.5458968
Time:
[1]  0.9611902 -0.4173185  0.9617922  2.1511244
Time:
[1] -1.11051
Time:
 [1]  0.15730149 -0.50623731  0.51752350  0.03958590 -0.60544167  0.25482640
 [7] -0.04631731 -0.79043621  1.10463205  0.60796246
Time:
[1] -0.6145621 -0.9958455  0.6387743 -2.2597655
[1] 0.1848823
[1] 15164086836
[1] 0.702374
[1] 15164086836
[1] 15164086837
